src/tiled/automappingutils.cpp

```diff
@@ -42,7 +43,10 @@ void eraseRegionObjectGroup(MapDocument *mapDocument,
         // tile objects. polygons and polylines are not covered correctly by this
         // erase method (we are in fact deleting too many objects)
         // TODO2: toAlignedRect may even break rects.
-        if (where.intersects(obj->bounds().toAlignedRect()))
+		
+		// Note: intersects returns false in some conditions so check if corners are contained 
+        const QRect& objRect = obj->bounds().toAlignedRect();
+        if (where.intersects(objRect) || where.contains(objRect.topLeft()) || where.contains(objRect.bottomLeft()))
```

Comment:
Hmm, `intersects` should only return false if the intersection is empty. This happens also when either rectangle is empty, even though one is contained in the other. So, it should be enough to check only one corner.